src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/ModellingGroupControllersTests.kt

```diff
@@ -118,10 +122,97 @@ class ModellingGroupControllersTests : ControllerTests<ModellingGroupController>
     }
 
     @Test
+    fun `getCoverageData returns long format if format=long`()
+    {
+        val repo = makeRepoMockingGetCoverageData(TouchstoneStatus.IN_PREPARATION)
+        val context = mock<ActionContext> {
+            on { it.params(":group-id") } doReturn "gId"
+            on { it.params(":touchstone-id") } doReturn "tId"
+            on { it.params(":scenario-id") } doReturn "sId"
+            on { it.queryParams("format") } doReturn "long"
+            on { hasPermission(any()) } doReturn true
+        }
+
+        val controller = ModellingGroupController(mockControllerContext())
+        val data = controller.getCoverageData(context, repo).data
+        Assertions.assertThat(data.first() is LongCoverageRow).isTrue()
+    }
+
+    @Test
+    fun `getCoverageData returns long format as default`()
+    {
+        val repo = makeRepoMockingGetCoverageData(TouchstoneStatus.IN_PREPARATION)
+        val context = mockContextForSpecificResponsibility(true)
+
+        val controller = ModellingGroupController(mockControllerContext())
+        val data = controller.getCoverageData(context, repo).data
+
+        // test data includes 4 years
+        // there are 7 other variable properties, test data includes 2 of each
+        val expectedRowCount = 4 * Math.pow(2.toDouble(), 7.toDouble())
+
+        Assertions.assertThat(data.first() is LongCoverageRow).isTrue()
+        Assertions.assertThat(data.count()).isEqualTo(expectedRowCount.toInt())
+    }
+
+    @Test
+    fun `getCoverageData returns wide format if format=wide`()
+    {
+        val repo = makeRepoMockingGetCoverageData(TouchstoneStatus.IN_PREPARATION)
+        val context = mock<ActionContext> {
+            on { it.params(":group-id") } doReturn "gId"
+            on { it.params(":touchstone-id") } doReturn "tId"
+            on { it.params(":scenario-id") } doReturn "sId"
+            on { it.queryParams("format") } doReturn "wide"
+            on { hasPermission(any()) } doReturn true
+        }
+
+        val controller = ModellingGroupController(mockControllerContext())
+        val data = controller.getCoverageData(context, repo).data
+
+        Assertions.assertThat(data.first() is WideCoverageRow).isTrue()
+
+        // there are 7 variable properties, test data includes 2 of each
+        val expectedRowCount = Math.pow(2.toDouble(), 7.toDouble())
+        Assertions.assertThat(data.count()).isEqualTo(expectedRowCount.toInt())
+
+        val expectedHeaders = listOf("1985_coverage", "1990_coverage", "1995_coverage", "2000_coverage",
+                "1985_target", "1990_target", "1995_target", "2000_target")
+
+        val firstRow = (data.first() as WideCoverageRow)
+        val values = firstRow.coverageAndTargetPerYear.values
+        val headers = firstRow.coverageAndTargetPerYear.keys
```

Comment:
I think this is not a very clear way of checking that the long format data has been correctly transformed. Perhaps you could explicitly check the coverage and target values are as expected for one of the rows and one of the years? Obviously that would need to be in combination with the deterministic change I suggested below. Otherwise, so long as the coverage and target values are different this test will pass, even if they are e.g. the wrong way around.